docs/source-2.0/additional-specs/rules-engine/specification.rst

```diff
@@ -74,6 +74,184 @@ or :ref:`error rules, <rules-engine-endpoint-rule-set-error-rule>` with an
 empty set of conditions to provide a more meaningful default or error depending
 on the scenario.
 
+.. smithy-trait:: smithy.rules#endpointBdd
+.. _smithy.rules#endpointBdd-trait:
+
+----------------------------------
+``smithy.rules#endpointBdd`` trait
+----------------------------------
+
+.. warning:: Experimental
+
+    This trait is experimental and subject to change.
+
+Summary
+    Defines a `Binary Decision Diagram (BDD) <https://en.wikipedia.org/wiki/Binary_decision_diagram>`_ representation
+    of endpoint rules for efficient runtime evaluation.
```

Comment:
May want to add something about compact storage/representation as well here (I know its mentioned later, but I think worth highlighting both benefits up front).